python/ray/tests/test_dataclient_disconnect.py

```diff
@@ -48,6 +59,7 @@ def f():
         assert not ray.is_connected()
 
         # Test that a new connection can be made
+        time.sleep(5)  # Give server time to clean up old connection
```

Comment:
Why this sleep is necessary?

Reply:
+1 (and also the one above). It seems like this could become a source of flakiness :)

Reply:
Needed since the client is shut down ungracefully (without notifying the server it could clean up the connection), and the server will wait 5 seconds (set by env var in the `with`) for the grace period to pass. Can revisit if it becomes an issue in the future, but this is about equivalent to how the test functioned before, just with the grace period introduced.

Reply:
Maybe sleep here more than 5 seconds just to make sure?